modules/demo/src/main/scala/lucuma/ui/demo/Demo.scala

```diff
@@ -289,27 +293,33 @@ trait AppMain extends IOApp {
         None
       )
 
-    IO {
-      val container = Option(dom.document.getElementById("root")).getOrElse {
-        val elem = dom.document.createElement("div")
-        elem.id = "root"
-        dom.document.body.appendChild(elem)
-        elem
-      }
+    Dispatcher[IO].allocated.map(_._1).flatMap { implicit dispatcher =>
```

Comment:
allocated?

Reply:
Unlike `use`, `allocated` doesn't close the resource upon de-scoping. Instead, it provides a callback to manually close it, which we happily ignore here (hence the `.map(_._1)`).

If we used `use` instead, the `dispatcher` would be rendered useless as soon as the `use` clause ended.

In a real-world app, we will probably want to store the close callback somewhere, and invoke it when the app is hot reloaded in dev (which will probably create another `Dispatcher` for the new iteration). In production this won't happen, so it's not an issue there.

Reply:
Agree it probably doesn't matter since this is a Demo app, but I think I'm missing something. This is the main IO of the app, right? When it exits it returns `ExitCode.Success` and isn't that the end of the application? If so, it seems like `use` would work fine.  Or, it it something about running in the browser I'm forgetting.

Reply:
The main method mounts the react app (`renderIntoDOM`) and exits immediately.